webui/react/src/components/TaskBar.tsx

```diff
@@ -66,19 +75,26 @@ export const TaskBar: React.FC<Props> = ({
   const menuItems: MenuItem[] = useMemo(
     () => [
       {
-        disabled: !canModifyWorkspaceNSC({ workspace: { id: task.workspaceId } }),
+        disabled: Loadable.match(task, {
+          _: () => true,
+          Loaded: (t) => !canModifyWorkspaceNSC({ workspace: { id: t.workspaceId } }),
+        }),
         key: MenuKey.Kill,
         label: 'Kill',
       },
       { key: MenuKey.ViewLogs, label: 'View Logs' },
     ],
-    [canModifyWorkspaceNSC, task.workspaceId],
+    [canModifyWorkspaceNSC, task],
   );
 
   const handleDropdown = (key: string) => {
+    if (!task) return;
     switch (key) {
       case MenuKey.Kill:
-        deleteTask(task);
+        Loadable.match(task, {
+          _: () => null,
+          Loaded: (t) => deleteTask(t),
+        });
```

Comment:
nit: given that we're not doing anything in the `NotLoaded | Errored` case, we can use `forEach` here:
```suggestion
        Loadable.forEach(task, deleteTask);
```